miden-lib/src/notes/mod.rs

```diff
@@ -4,96 +4,104 @@ use miden_objects::{
     assets::Asset,
     notes::{Note, NoteScript},
     utils::{collections::Vec, vec},
-    Digest, Felt, Hasher, NoteError, Word, ZERO,
+    Felt, NoteError, Word, ZERO,
 };
 
+use miden_objects::crypto::rand::FeltRng;
+
 use super::transaction::TransactionKernel;
 
-// STANDARDIZED SCRIPTS
+mod utils;
+
+// STANDARDIZED SCRIPTS NOTE GENERATORS
 // ================================================================================================
 
-pub enum Script {
-    P2ID { target: AccountId },
-    P2IDR { target: AccountId, recall_height: u32 },
-    SWAP { asset: Asset, serial_num: Word },
+/// Generates a P2ID - pay to id note.
+pub fn create_p2id_note<R: FeltRng>(
+    sender: AccountId,
+    target: AccountId,
+    assets: Vec<Asset>,
+    mut rng: R,
+) -> Result<Note, NoteError> {
+    let assembler = TransactionKernel::assembler();
+    let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/P2ID.masb"));
+    let script_ast = ProgramAst::from_bytes(bytes).map_err(NoteError::NoteDeserializationError)?;
+    let inputs = vec![target.into(), ZERO, ZERO, ZERO];
+    let (note_script, _) = NoteScript::new(script_ast, &assembler)?;
+    let tag: Felt = target.into();
+    let serial_num = rng.draw_word();
+    Note::new(note_script.clone(), &inputs, &assets, serial_num, sender, tag)
```

Comment:
I would arrange the code slightly differently to make it a bit easier to follow. For example:
```Rust
let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/P2ID.masb"));
let note_script = build_note_script(bytes);

let inputs = vec![target.into(), ZERO, ZERO, ZERO];
let tag: Felt = target.into();
let serial_num = rng.draw_word();

Note::new(note_script, &inputs, &assets, serial_num, sender, tag)
```
Where `build_note_script()` is a small helper function. I'd apply the same to other notes below.